pyvista/core/common.py

```diff
@@ -1070,3 +1070,98 @@ def find_closest_point(self, point, n=1):
+
+        Returns
+        -------
+        index : int or np.ndarray
+            Index or indices of the cell in this mesh that is closest
+            to the given point.
+
+        Examples
+        --------
+        Find nearest cell to a point on a sphere
+
+        >>> import pyvista
+        >>> mesh = pyvista.Sphere()
+        >>> index = mesh.find_closest_cell([0, 0, 0.5])
+        >>> index
+        59
+
+        Find the nearest cells to several random points.  Note that
+        ``-1`` indicates that the locator was not able to find a
+        reasonably close cell.
+
+        >>> points = np.random.random((1000, 3))
+        >>> indices = mesh.find_closest_cell(points)
+        >>> print(indices.shape)
+        (1000,)
+        """
+        if not isinstance(point, (np.ndarray, collections.abc.Sequence)):
+            raise TypeError("Given point be an iterable or an array.")
+
+        # check if this is an array of points
+        is_array = False
+        if isinstance(point, np.ndarray):
+            if point.ndim > 2:
+                raise ValueError("Array of points must be 2D")
+            if point.ndim == 2:
+                if point.shape[1] != 3:
+                    raise ValueError("Array of points must have three values per point")
+                is_array = True
+            else:
+                if point.size != 3:
+                    raise ValueError("Given point must have three values")
+        else:
+            if len(point) != 3:
+                raise ValueError("Given point must have three values")
+
+        locator = vtk.vtkCellLocator()
+        locator.SetDataSet(self)
+        locator.BuildLocator()
+        if is_array:  # simply iterate through all the points
+            return np.array([locator.FindCell(node) for node in point])
+
+        return locator.FindCell(point)
+
+
+def axis_rotation(points, angle, inplace=False, deg=True, axis='z'):
+    """Rotate points angle (in deg) about an axis."""
+    axis = axis.lower()
+
+    # Copy original array to if not inplace
+    if not inplace:
+        points = points.copy()
+
+    # Convert angle to radians
+    if deg:
+        angle *= np.pi / 180
+
+    if axis == 'x':
+        y = points[:, 1] * np.cos(angle) - points[:, 2] * np.sin(angle)
+        z = points[:, 1] * np.sin(angle) + points[:, 2] * np.cos(angle)
+        points[:, 1] = y
+        points[:, 2] = z
+    elif axis == 'y':
+        x = points[:, 0] * np.cos(angle) + points[:, 2] * np.sin(angle)
+        z = - points[:, 0] * np.sin(angle) + points[:, 2] * np.cos(angle)
+        points[:, 0] = x
+        points[:, 2] = z
+    elif axis == 'z':
+        x = points[:, 0] * np.cos(angle) - points[:, 1] * np.sin(angle)
+        y = points[:, 0] * np.sin(angle) + points[:, 1] * np.cos(angle)
+        points[:, 0] = x
+        points[:, 1] = y
+    else:
+        raise ValueError('invalid axis. Must be either "x", "y", or "z"')
+
+    if not inplace:
+        return points
```

Comment:
This was moved to a utilities file no?

Reply:
I'll remove that.  Thanks for catching that.